### DISCUSSION TITLE
React Native

### DISCUSSION THREAD

**Discussion by luigiminardim** [mdx-js/mdx#1355] [Q&A] [answered]
I couldn't find step by step showing how to use MDX in React Native applications. I thought that it was just needed to install `yarn add @mdx-js/react`. I'm using **expo**, so, if anyone can help, I would appreciate it.

**Comment by ChristianMurphy** [comment, marked as answer]
There's some discussion on usage in react native here: https://github.com/mdx-js/mdx/issues/972
Specifically https://github.com/mdx-js/mdx/issues/972#issuecomment-706358772

**Reply by luigiminardim** [reply]
Thanks for the comment. But my question is more related to how to integrate create-react-native-app with MDX, already adding typescript. The documentation only clarifies how to integrate MDX into a React project without a typescript.

What I've tried to do was:
1. In terminal: `yarn add mdx.macro`;
2. Add, in my _app.d.ts_ file, all the content from [the documentation typescript section](https://mdxjs.com/advanced/typescript);
3. Add this code in my App.tsx: 
  ```
  import {importMdx} from  'mdx.macro'
  const Content = lazy(() => importMDX('./Content.mdx')))
  ```
4. I get this error:  
  ```
  Could not find a declaration file for module 'mdx.macro'. 
  '/home/luigiminardim/Desktop/HRTRoutineProject/node_modules/mdx.macro/mdx.macro.js' implicitly has an 'any' type. 
  ```

I was expecting to solve my problem just by `yarn add @types/mdx.macro`, but, apparently, there is no such types declarations package.

**Reply by ChristianMurphy** [reply]
mdx macro is a community package, not maintained by mdx https://github.com/frontarm/mdx-util
it's behind by a few releases and in need of a rewrite https://github.com/mdx-js/mdx/pull/589#issuecomment-499325405

I'd recommend using `@mdx-js/loader` instead.
Which is maintained and documented as part of mdx https://mdxjs.com/getting-started/webpack
and can also be used inline if/when needed.

```jsx
import Content from '!babel-loader!@mdx-js/loader!./Content.mdx'
```

version 2 of MDX includes types https://github.com/mdx-js/mdx/issues/1041, including documentation/examples of typing mdx imports.
